drakrun/drakrun/draksetup.py

```diff
@@ -390,44 +393,38 @@ def send_usage_report(report):
-        except subprocess.CalledProcessError:
-            raise RuntimeError(f"Failed to mount {block_device} as NTFS.")
+            logging.info(f"Fetching rekall profile for {file.path}")
 
-        for file in dll_file_list:
-            try:
-                logging.info(f"Fetching rekall profile for {file.path}")
-                local_dll_path = os.path.join(PROFILE_DIR, file.dest)
-
-                copyfile(os.path.join(mount_path, file.path), local_dll_path)
-                guid = pdb_guid(local_dll_path)
-                tmp = fetch_pdb(guid["filename"], guid["GUID"], PROFILE_DIR)
-
-                logging.debug("Parsing PDB into JSON profile...")
-                profile = make_pdb_profile(tmp)
-                with open(os.path.join(PROFILE_DIR, f"{file.dest}.json"), 'w') as f:
-                    f.write(profile)
-            except FileNotFoundError:
-                logging.warning(f"Failed to copy file {file.path}, skipping...")
-            except RuntimeError:
-                logging.warning(f"Failed to fetch profile for {file.path}, skipping...")
-            except Exception:
-                logging.warning(f"Unexpected exception while creating rekall profile for {file.path}, skipping...")
-            finally:
-                if os.path.exists(local_dll_path):
-                    os.remove(local_dll_path)
-                if os.path.exists(os.path.join(PROFILE_DIR, tmp)):
-                    os.remove(os.path.join(PROFILE_DIR, tmp))
-
-        # cleanup
-        subprocess.check_output(f'umount {mnt_path_quoted}', shell=True)
+            local_dll_path = os.path.join(PROFILE_DIR, file.dest)
+            guest_dll_path = str(PureWindowsPath("C:/", file.path))
+
+            injector.read_file(guest_dll_path, local_dll_path)
+            guid = pdb_guid(local_dll_path)
+            tmp = fetch_pdb(guid["filename"], guid["GUID"], PROFILE_DIR)
+
+            logging.debug("Parsing PDB into JSON profile...")
+            profile = make_pdb_profile(tmp)
+            with open(os.path.join(PROFILE_DIR, f"{file.dest}.json"), 'w') as f:
+                f.write(profile)
+        except subprocess.CalledProcessError:
+            logging.warning(f"Failed to copy file {file.path}, skipping...")
+        except RuntimeError:
+            logging.warning(f"Failed to fetch profile for {file.path}, skipping...")
+        except Exception:
+            # Can help in debugging
+            logging.warning(f"Unexpected exception while creating rekall profile for {file.path}, skipping...")
+            logging.debug(traceback.format_exc())
+        finally:
+            if os.path.exists(local_dll_path):
+                os.remove(local_dll_path)
+            # was crashing here if the first file reached some exception
+            if 'tmp' in locals() and os.path.exists(os.path.join(PROFILE_DIR, tmp)):
+                os.remove(os.path.join(PROFILE_DIR, tmp))
```

Comment:
Instead of this, I think it would be simpler to initialize `tmp` with `None` and then check `if tmp is not None and os.path.exists(..)`